server/safemapstructure/safemapstructure.go

```diff
@@ -0,0 +1,20 @@
+package safemapstructure
+
+import (
+	"github.com/mitchellh/mapstructure"
+)
+
+func Decode(input interface{}, output interface{}) error {
+	config := &mapstructure.DecoderConfig{
+		Metadata:  nil,
+		Result:    output,
+		MatchName: func(a string, b string) bool { return a == b }, // Only match exactly
```

Comment:
Exact matches are handled in a different code path so `MatchName` never gets called with equal values of `a` and `b`. Which means this could also be written as:

```suggestion
		MatchName: func(string, string) bool { return false }, // Only match exactly
```

0/5 on whether that's better or worse than what you've got here.

Reply:
I did notice that. I just thought that this was more clear.